citesphere/src/main/webapp/WEB-INF/views/auth/group/editItem.jsp

```diff
@@ -73,6 +80,58 @@ $(function() {
 				$("#editForm").append(affiliationIdField);
 			});
 		});
+		$(".editor-item").each(function(idx, editor) {
+			var editorIdField = $("<input>");
+			editorIdField.attr("type", "hidden");
+			editorIdField.attr("id", "editors" + idx + ".id");
+			editorIdField.attr("name", "editors[" + idx + "].id");
+			editorIdField.attr("value", $(editor).data("editor-id"));
+			$("#editForm").append(editorIdField);
+			
+			var editorFirstNameField = $("<input>");
+			editorFirstNameField.attr("type", "hidden");
+			editorFirstNameField.attr("id", "editors" + idx + ".firstName");
+			editorFirstNameField.attr("name", "editors[" + idx + "].firstName");
+			editorFirstNameField.attr("value", $(editor).data("editor-firstname"));
+			$("#editForm").append(editorFirstNameField);
+			
+			var editorLastNameField = $("<input>");
+			editorLastNameField.attr("type", "hidden");
+			editorLastNameField.attr("id", "editors" + idx + ".lastName");
+			editorLastNameField.attr("name", "editors[" + idx + "].lastName");
+			editorLastNameField.attr("value", $(editor).data("editor-lastname"));
+			$("#editForm").append(editorLastNameField);
+			
+			var editorUriField = $("<input>");
+			editorUriField.attr("type", "hidden");
+			editorUriField.attr("id", "editors" + idx + ".uri");
+			editorUriField.attr("name", "editors[" + idx + "].uri");
+			editorUriField.attr("value", $(editor).data("editor-uri"));
+			$("#editForm").append(editorUriField);
+			
+			var editorAuthorityField = $("<input>");
+			editorAuthorityField.attr("type", "hidden");
+			editorAuthorityField.attr("id", "editors" + idx + ".localAuthorityId");
+			editorAuthorityField.attr("name", "editors[" + idx + "].localAuthorityId");
+			editorAuthorityField.attr("value", $(editor).data("editor-authority-id"));
+			$("#editForm").append(editorAuthorityField);
+			
+			$(editor).children("span").each(function(idx2, affiliation) {
+				var affiliationField = $("<input>");
+				affiliationField.attr("type", "hidden");
+				affiliationField.attr("id", "editors" + idx + ".affiliations" + idx2 + ".name");
+				affiliationField.attr("name", "editors[" + idx + "].affiliations[" + idx2 + "].name");
+				affiliationField.attr("value", $(affiliation).data("affiliation-name"));
+				$("#editForm").append(affiliationField);
+				
+				var affiliationIdField = $("<input>");
+				affiliationIdField.attr("type", "hidden");
+				affiliationIdField.attr("id", "editors" + idx + ".affiliations" + idx2 + ".id");
+				affiliationIdField.attr("name", "editors[" + idx + "].affiliations[" + idx2 + "].id");
+				affiliationIdField.attr("value", $(affiliation).data("affiliation-id"));
+				$("#editForm").append(affiliationIdField);
+			});
+		});
```

Comment:
let's create a function that takes a span element (of editor or author) and a array name ("editors" or "authors") and then put all of this into the method and instead of hard-coding 'authors' or 'editors' when adding id and name attributes use the past array name. does this make sense?

Reply:
And then you can also replace all these blocks:
`
var affiliationField = $("<input>");
affiliationField.attr("type", "hidden");
affiliationField.attr("id", "editors" + idx + ".affiliations" + idx2 + ".name");
affiliationField.attr("name", "editors[" + idx + "].affiliations[" + idx2 + "].name");
affiliationField.attr("value", $(affiliation).data("affiliation-name"));
				$("#editForm").append(affiliationField);
`
(or the equivalent editor/author blocks) with a method that takes the property name (in the above example "name") and the data field that needs to be accessed (e.g. "affiliation-name").